lib/index.d.ts

```diff
@@ -10,20 +10,24 @@ export interface Action<P> extends AnyAction {
     error?: boolean;
     meta?: Meta;
 }
-export declare type RequiredKeys<T> = {
-    [P in keyof T]: T[P] extends undefined ? never : P;
-}[keyof T];
-export declare type Optionalize<T> = Partial<T> & {
```

Comment:
I think adding the following:

```ts
type OptionalKeys<T> = Exclude<keyof T, RequiredKeys<T>>;
```

And changing `Optionalize<T>` to:

```ts
type Optionalize<T> = {
    [P in RequiredKeys<T>]: T[P];
} & {
    [P in OptionalKeys<T>]?: T[P];
};
```

Will do what you want but also be a little clearer?